### ISSUE TITLE
Ticket Templates: Dirty fields don't cause draft state in new ticket dialogue

### ISSUE THREAD

**Issue by MrGeneration** [zammad/zammad#4375]
### Infos:

* Used Zammad version: develop / 5.3
* Installation method (source, package, ..): any
* Operating system: any
* Database + version: any
* Elasticsearch version: any
* Browser + version: verified on latest Firefox (Windows)

### Expected behavior:

When causing dirty fields (pre filled fields) followed by an applied template, Zammad will make sure you really want to discard your draft.

### Actual behavior:

When causing dirty fields (pre filled fields) followed by an applied template, Zammad will silently close your draft.
This does not happen if you ensure to "dirty a field" after applying the template.

![dirty-fields-are-not-that-dirty](https://user-images.githubusercontent.com/6549061/202995776-26a82109-51d4-467d-af6f-71f0ae4d1536.gif)

### Steps to reproduce the behavior:

* create a ticket tempalte with e.g. article text and ticket title (save)
* open a new ticket dialogue
* make either field dirty
* apply template
* try to close the new ticket dialogue tab

Yes I'm sure this is a bug and no feature request or a general question.

**Comment by dvuckovic** [comment]
I was not able to reproduce this with Zammad 5.3.0, the ticket create form will properly show the confirmation dialog if there was a field changed or template applied:

https://user-images.githubusercontent.com/6049445/206194228-989c4c24-e640-4beb-a654-a9390753d384.mp4

There are several points here:

- A field is considered dirty if there was a user interaction with it (i.e. user entered the field and typed something in it)
- A field is NOT considered dirty if only a template was applied, allowing for easy switching between templates (as explained in 3rd point [here](https://github.com/zammad/zammad/issues/2434#issuecomment-1109775937))
- A form is considered dirty whenever its state changes and taskbar mechanism saves it on the server (be that from the user interaction or when applying the template)
- The confirmation dialog is linked to the saved taskbar state, and will kick in only if it exists

Perhaps there was something wrong with the taskbar mechanism on the affected system? This is running in the background, and can be checked by looking for **POST** `/api/v1/taskbar/*` requests in the browser dev tools. Theoretically, if the taskbar response does not come back in time (i.e. the user is too fast), the form state might not be flagged as dirty in time. I was not able to reproduce this race condition locally, and I tried my best to be very fast. On the other hand, my system is quite small and lean, so YMMV.

**Comment by MrGeneration** [comment]
Yeah sorry to bomb the party. The described issue is still there as on a current pull from tonight. (2am).
Your screencast doesn't feature dirtying the field before applying the template, not sure if you have verified that.

Dirtying fields *after* applying is not a problem at all.
If you think I'm too fast, how long should I wait to allow Zammad to respond in a timely manner in your opinion?

![itsnotafeature](https://user-images.githubusercontent.com/6549061/206298823-c1578075-db69-46a0-a0df-f1d4454237e1.gif)

**Comment by mantas** [comment]
@MrGeneration are you able to reproduce this on a fresh system? How does the template look like?

I can't reproduce it either with following steps:

- fill in customer
- apply template with title OR title+customer OR title+article+customer
- taskbar closes successfully

**Comment by MrGeneration** [comment]
Yes this happens on a freshly seeded instance with added demo chrispresso. All of that from scratch.

> taskbar closes successfully

What do you mean "successfully" ?
Closing the ticket draft WITHOUT WARNING in this scenario is all but successful. I'm expecting the data loss warning.

I by now have verified this 3 times. How many times do I have to redo this?

**Comment by mantas** [comment]
I mean warning is shown.

How does your template look like?

**Comment by fliebe92** [comment]
@dvuckovic, @mantas > Did you try to reproduce this on a demo machine from Marcel? I guess it's somehow connected to it.
